utils/java-dynamic-memory-opts

```diff
@@ -14,10 +14,19 @@ if [ -z "$MEM_JAVA_PERCENT" ]; then
     fi
 fi
 
-# MEM_TOTAL_KB can be set from the outside, this is important for container-based infrastructure:
-# /proc/meminfo would return the host/node capacity and not the container's limit
+# Take the lower of /sys/fs/cgroup/memory/memory.limit_in_bytes and
+# /proc/meminfo. This takes into account memory limits set by containers
+# Setting MEM_TOTAL_KB is still supported
 if [ -z "$MEM_TOTAL_KB" ]; then
-    MEM_TOTAL_KB=$(cat /proc/meminfo | grep MemTotal | awk '{print $2}')
+    MEM_LIMIT_CGROUP=$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes 2>/dev/null || echo 9223372036854771712)
```

Comment:
What is `9223372036854771712` here in the echo?

Reply:
That is the maximum value for this particular sysfs node, this is also displayed when the memory is not limited by cgroup.

Reply:
I think it could be documented in the code e.g. through a comment.